x-pack/plugins/index_lifecycle_management/public/application/sections/policy_table/components/policy_action.tsx

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { usePolicyListContext } from '../policy_list_context';
+import { IndexTemplatesFlyout } from '../../../components/index_templates_flyout';
+import { ConfirmDelete } from './confirm_delete';
+import { AddPolicyToTemplateConfirmModal } from './add_policy_to_template_confirm_modal';
+
+interface Props {
+  updatePolicies: () => void;
+}
+export const PolicyAction: React.FunctionComponent<Props> = ({ updatePolicies }) => {
+  const { policyAction, setPolicyAction } = usePolicyListContext();
+  if (policyAction?.action === 'viewIndexTemplates') {
+    return (
+      <IndexTemplatesFlyout
+        policy={policyAction.policy}
+        close={() => {
+          setPolicyAction(null);
+        }}
+      />
+    );
+  }
+  if (policyAction?.action === 'deletePolicy') {
+    return (
+      <ConfirmDelete
+        policyToDelete={policyAction.policy}
+        callback={() => {
+          updatePolicies();
+          setPolicyAction(null);
+        }}
+        onCancel={() => {
+          setPolicyAction(null);
+        }}
+      />
+    );
+  }
+
+  if (policyAction?.action === 'addIndexTemplate') {
+    return (
+      <AddPolicyToTemplateConfirmModal
+        policy={policyAction.policy}
+        onSuccess={(indexTemplate: string) => {
+          policyAction!.policy.indexTemplates = [
```

Comment:
Can you explain the expected behavior here? Shouldn't `setPolicyAction` be used to update the policy?

Reply:
I added a comment about updating index templates list